File: typescript/src/models/ServiceList.ts
/* tslint:disable */
/* eslint-disable */
/*
 * Cloud Governance Api
 *
 * Contact: support@avepoint.com
 */

import { exists, mapValues } from '../runtime';
import {
    ServiceType,
    ServiceTypeFromJSON,
    ServiceTypeFromJSONTyped,
    ServiceTypeToJSON,
} from './';

/**
 * 
 * @export
 * @interface ServiceList
 */
export interface ServiceList {
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    id?: string;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    name?: string | null;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    description?: string | null;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    category?: string | null;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    contact?: string | null;
    /**
     * 
     * @type {number}
     * @memberof ServiceList
     */
    languageId?: number;
    /**
     * 
     * @type {ServiceType}
     * @memberof ServiceList
     */
    type?: ServiceType;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    typeDescription?: string | null;
    /**
     * 
     * @type {boolean}
     * @memberof ServiceList
     */
    isQuestionnaire?: boolean;
    /**
     * 
     * @type {string}
     * @memberof ServiceList
     */
    questionnaireId?: string | null;
}

export function ServiceListFromJSON(json: any): ServiceList {
    return ServiceListFromJSONTyped(json, false);
}

export function ServiceListFromJSONTyped(json: any, ignoreDiscriminator: boolean): ServiceList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'category': !exists(json, 'category') ? undefined : json['category'],
        'contact': !exists(json, 'contact') ? undefined : json['contact'],
        'languageId': !exists(json, 'languageId') ? undefined : json['languageId'],
        'type': !exists(json, 'type') ? undefined : ServiceTypeFromJSON(json['type']),
        'typeDescription': !exists(json, 'typeDescription') ? undefined : json['typeDescription'],
        'isQuestionnaire': !exists(json, 'isQuestionnaire') ? undefined : json['isQuestionnaire'],
        'questionnaireId': !exists(json, 'questionnaireId') ? undefined : json['questionnaireId'],
    };
}

export function ServiceListToJSON(value?: ServiceList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'id': value.id,
        'name': value.name,
        'description': value.description,
        'category': value.category,
        'contact': value.contact,
        'languageId': value.languageId,
        'type': ServiceTypeToJSON(value.type),
        'typeDescription': value.typeDescription,
        'isQuestionnaire': value.isQuestionnaire,
        'questionnaireId': value.questionnaireId,
    };
}

File: csharp-netstandard/src/Cloud.Governance.Client/Model/ServiceList.cs


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Cloud.Governance.Client.Client.OpenAPIDateConverter;

namespace Cloud.Governance.Client.Model
{
    /// <summary>
    /// ServiceList
    /// </summary>
    [DataContract(Name = "ServiceList")]
    public partial class ServiceList : IEquatable<ServiceList>, IValidatableObject
    {
        /// <summary>
        /// Gets or Sets Type
        /// </summary>
        [DataMember(Name = "type", EmitDefaultValue = false)]
        public ServiceType? Type { get; set; }
        /// <summary>
        /// Initializes a new instance of the <see cref="ServiceList" /> class.
        /// </summary>
        /// <param name="id">id.</param>
        /// <param name="name">name.</param>
        /// <param name="description">description.</param>
        /// <param name="category">category.</param>
        /// <param name="contact">contact.</param>
        /// <param name="languageId">languageId (default to 0).</param>
        /// <param name="type">type.</param>
        /// <param name="typeDescription">typeDescription.</param>
        /// <param name="isQuestionnaire">isQuestionnaire (default to false).</param>
        /// <param name="questionnaireId">questionnaireId.</param>
        public ServiceList(Guid id = default(Guid), string name = default(string), string description = default(string), string category = default(string), string contact = default(string), int languageId = 0, ServiceType? type = default(ServiceType?), string typeDescription = default(string), bool isQuestionnaire = false, Guid? questionnaireId = default(Guid?))
        {
            this.Id = id;
            this.Name = name;
            this.Description = description;
            this.Category = category;
            this.Contact = contact;
            this.LanguageId = languageId;
            this.Type = type;
            this.TypeDescription = typeDescription;
            this.IsQuestionnaire = isQuestionnaire;
            this.QuestionnaireId = questionnaireId;
        }

        /// <summary>
        /// Gets or Sets Id
        /// </summary>
        [DataMember(Name = "id", EmitDefaultValue = false)]
        public Guid Id { get; set; }

        /// <summary>
        /// Gets or Sets Name
        /// </summary>
        [DataMember(Name = "name", EmitDefaultValue = true)]
        public string Name { get; set; }

        /// <summary>
        /// Gets or Sets Description
        /// </summary>
        [DataMember(Name = "description", EmitDefaultValue = true)]
        public string Description { get; set; }

        /// <summary>
        /// Gets or Sets Category
        /// </summary>
        [DataMember(Name = "category", EmitDefaultValue = true)]
        public string Category { get; set; }

        /// <summary>
        /// Gets or Sets Contact
        /// </summary>
        [DataMember(Name = "contact", EmitDefaultValue = true)]
        public string Contact { get; set; }

        /// <summary>
        /// Gets or Sets LanguageId
        /// </summary>
        [DataMember(Name = "languageId", EmitDefaultValue = false)]
        public int LanguageId { get; set; }

        /// <summary>
        /// Gets or Sets TypeDescription
        /// </summary>
        [DataMember(Name = "typeDescription", EmitDefaultValue = true)]
        public string TypeDescription { get; set; }

        /// <summary>
        /// Gets or Sets IsQuestionnaire
        /// </summary>
        [DataMember(Name = "isQuestionnaire", EmitDefaultValue = false)]
        public bool IsQuestionnaire { get; set; }

        /// <summary>
        /// Gets or Sets QuestionnaireId
        /// </summary>
        [DataMember(Name = "questionnaireId", EmitDefaultValue = true)]
        public Guid? QuestionnaireId { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ServiceList {\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Name: ").Append(Name).Append("\n");
            sb.Append("  Description: ").Append(Description).Append("\n");
            sb.Append("  Category: ").Append(Category).Append("\n");
            sb.Append("  Contact: ").Append(Contact).Append("\n");
            sb.Append("  LanguageId: ").Append(LanguageId).Append("\n");
            sb.Append("  Type: ").Append(Type).Append("\n");
            sb.Append("  TypeDescription: ").Append(TypeDescription).Append("\n");
            sb.Append("  IsQuestionnaire: ").Append(IsQuestionnaire).Append("\n");
            sb.Append("  QuestionnaireId: ").Append(QuestionnaireId).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as ServiceList);
        }

        /// <summary>
        /// Returns true if ServiceList instances are equal
        /// </summary>
        /// <param name="input">Instance of ServiceList to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ServiceList input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.Id == input.Id ||
                    (this.Id != null &&
                    this.Id.Equals(input.Id))
                ) && 
                (
                    this.Name == input.Name ||
                    (this.Name != null &&
                    this.Name.Equals(input.Name))
                ) && 
                (
                    this.Description == input.Description ||
                    (this.Description != null &&
                    this.Description.Equals(input.Description))
                ) && 
                (
                    this.Category == input.Category ||
                    (this.Category != null &&
                    this.Category.Equals(input.Category))
                ) && 
                (
                    this.Contact == input.Contact ||
                    (this.Contact != null &&
                    this.Contact.Equals(input.Contact))
                ) && 
                (
                    this.LanguageId == input.LanguageId ||
                    this.LanguageId.Equals(input.LanguageId)
                ) && 
                (
                    this.Type == input.Type ||
                    this.Type.Equals(input.Type)
                ) && 
                (
                    this.TypeDescription == input.TypeDescription ||
                    (this.TypeDescription != null &&
                    this.TypeDescription.Equals(input.TypeDescription))
                ) && 
                (
                    this.IsQuestionnaire == input.IsQuestionnaire ||
                    this.IsQuestionnaire.Equals(input.IsQuestionnaire)
                ) && 
                (
                    this.QuestionnaireId == input.QuestionnaireId ||
                    (this.QuestionnaireId != null &&
                    this.QuestionnaireId.Equals(input.QuestionnaireId))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.Id != null)
                    hashCode = hashCode * 59 + this.Id.GetHashCode();
                if (this.Name != null)
                    hashCode = hashCode * 59 + this.Name.GetHashCode();
                if (this.Description != null)
                    hashCode = hashCode * 59 + this.Description.GetHashCode();
                if (this.Category != null)
                    hashCode = hashCode * 59 + this.Category.GetHashCode();
                if (this.Contact != null)
                    hashCode = hashCode * 59 + this.Contact.GetHashCode();
                hashCode = hashCode * 59 + this.LanguageId.GetHashCode();
                hashCode = hashCode * 59 + this.Type.GetHashCode();
                if (this.TypeDescription != null)
                    hashCode = hashCode * 59 + this.TypeDescription.GetHashCode();
                hashCode = hashCode * 59 + this.IsQuestionnaire.GetHashCode();
                if (this.QuestionnaireId != null)
                    hashCode = hashCode * 59 + this.QuestionnaireId.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
